### DISCUSSION TITLE
3.0.0-beta.3

### DISCUSSION THREAD

**Discussion by zHaytam** [Blazor-Diagrams/Blazor.Diagrams#246] [Announcements]
### Added

- Support for `LinkFactory` to return null in order to not create an ongoing link
- Support for free links (no source/target required)
- `PositionAnchor` which reprensents a simple plain position (mutable)
- `ArrowHeadControl` to control a link's Source/Target on the fly
- `attached` css class to attached links

### Changed

- Replace `OngoingPosition` with the new `PositionAnchor`
	- `BaseLinkModel.Target` will never be null anymore. An ongoing link will have a position anchor as the target
- `Links.Factory` signature now takes the diagram, source (model) and the target anchor
- Move `DynamicAnchor` back to `Anchors` namespace and seal all `Anchor` classes

### Fixed

- Links attached to links not refreshing when the others are
- `LinkPathPositionProvider` not working with maxlength ratios
- Deleting a link not deleting the links attached to it

### Removed

- `PositionProvider` argument from `ExecutableControl` for more freedom
- `Id` and `Refresh` from `ILinkable`
- Unused `Offset` from `Anchor` and make `Model` nullable

<hr /><em>This discussion was created from the release <a href='https://github.com/Blazor-Diagrams/Blazor.Diagrams/releases/tag/3.0.0-beta.3'>3.0.0-beta.3</a>.</em>

**Comment by snehabihani** [comment]
Hi,

https://github.com/Blazor-Diagrams/Blazor.Diagrams/issues/216
Is this released?

**Reply by zHaytam** [reply]
The changes that you're interested in are still in progress: https://github.com/Blazor-Diagrams/Blazor.Diagrams/projects/7#card-85094622

**Reply by snehabihani** [reply]
Thank you for quick reply.
